components/email_aliases/email_aliases_service.h

```diff
@@ -39,13 +54,42 @@ class EmailAliasesService : public KeyedService,
   void AddObserver(mojo::PendingRemote<mojom::EmailAliasesServiceObserver>
                        observer) override;
 
+  // Response handlers
+  void OnRequestAuthenticationResponse(
+      RequestAuthenticationCallback callback,
+      std::optional<std::string> response_body);
+  void OnRequestSessionResponse(std::optional<std::string> response_body);
+
   // Binds the mojom interface to this service
   void BindInterface(
       mojo::PendingReceiver<mojom::EmailAliasesService> receiver);
 
+  std::string GetAuthTokenForTesting() const;
+
+  static std::string GetAccountsServiceVerifyInitURL();
+  static std::string GetAccountsServiceVerifyResultURL();
+
  private:
+  void ApiFetch(const GURL& url,
+                const char* method,
+                const std::optional<std::string>& bearer_token,
+                const base::Value::Dict& bodyValue,
+                BodyAsStringCallback download_to_string_callback);
+  void RequestSession();
+  void NotifyObserversAuthStateChanged(mojom::AuthenticationStatus status);
+  void ResetVerificationFlow();
+
   mojo::ReceiverSet<mojom::EmailAliasesService> receivers_;
   mojo::RemoteSet<mojom::EmailAliasesServiceObserver> observers_;
+  std::string verification_token_;
+  std::string auth_token_;
+  std::string auth_email_;
+  scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory_;
+  std::unique_ptr<network::SimpleURLLoader> verification_simple_url_loader_;
+  // Cached URLs computed once per service lifetime
+  std::string verify_init_url_;
+  std::string verify_result_url_;
+  base::WeakPtrFactory<EmailAliasesService> weak_factory_{this};
```

Comment:
We need better comments for this member fields, but also for the whole class in general, including top comment explaining what the class does... comments for the new functions individually, and comments for these fields.

For reference:

https://google.github.io/styleguide/cppguide.html#Class_Comments
https://google.github.io/styleguide/cppguide.html#Function_Comments
https://google.github.io/styleguide/cppguide.html#Variable_Comments